internal/ui/ui_glfw.go

```diff
@@ -203,38 +199,102 @@ func initialize() error {
 	return nil
 }
 
-type monitor struct {
+func (u *userInterfaceImpl) setInitMonitor(m *glfw.Monitor) {
+	u.initMonitor = m
+	u.initDeviceScaleFactor = u.deviceScaleFactor(m)
+	// GetVideoMode must be called from the main thread, then call this here and record
+	// initFullscreen{Width,Height}InDIP.
+	v := m.GetVideoMode()
+	u.initFullscreenWidthInDIP = int(u.dipFromGLFWMonitorPixel(float64(v.Width), m))
+	u.initFullscreenHeightInDIP = int(u.dipFromGLFWMonitorPixel(float64(v.Height), m))
+}
+
+// Monitor is a wrapper around glfw.Monitor.
+type Monitor struct {
 	m  *glfw.Monitor
 	vm *glfw.VidMode
 	// Pos of monitor in virtual coords
-	x int
-	y int
+	x      int
+	y      int
+	width  int
+	height int
+	id     int
+	name   string
+}
+
+// Bounds returns the monitor's bounds.
+func (m *Monitor) Bounds() image.Rectangle {
+	ui := Get()
+	return image.Rect(
+		int(ui.dipFromGLFWMonitorPixel(float64(m.x), m.m)),
+		int(ui.dipFromGLFWMonitorPixel(float64(m.y), m.m)),
+		int(ui.dipFromGLFWMonitorPixel(float64(m.x+m.width), m.m)),
+		int(ui.dipFromGLFWMonitorPixel(float64(m.x+m.height), m.m)),
+	)
+}
+
+// Name returns the monitor's name.
+func (m *Monitor) Name() string {
+	return m.name
 }
 
 // monitors is the monitor list cache for desktop glfw compile targets.
 // populated by 'updateMonitors' which is called on init and every
 // monitor config change event.
 //
 // monitors must be manipulated on the main thread.
-var monitors []*monitor
+var monitors []*Monitor
+
+// AppendMonitors appends the current monitors to the passed in mons slice and returns it.
+func (u *userInterfaceImpl) AppendMonitors(mons []*Monitor) []*Monitor {
+	u.m.RLock()
+	defer u.m.RUnlock()
+	return append(mons, monitors...)
+}
+
+// Monitor returns the window's current monitor. Returns nil if there is no current monitor yet.
+func (u *userInterfaceImpl) Monitor() *Monitor {
+	u.m.RLock()
+	defer u.m.RUnlock()
+	glfwMonitor := u.currentMonitor()
```

Comment:
Ahhh, I realized that this must be called on the main thread. So, like `ScreenSizeInFullscreen`, we have to have:

```go
func (u *userInterfaceImpl) Monitor() *Monitor {
    if !u.isRunning() {
         return /* initial monitor?? */
    }
    var monitor *Monitor
    u.mainThread.Call(func() {
        /* ... */
    })
    return monitor
}
```